src/_pytest/terminal.py

```diff
@@ -825,16 +825,30 @@ def summary_warnings(self):
             for wr in warning_reports:
                 reports_grouped_by_message.setdefault(wr.message, []).append(wr)
 
-            title = "warnings summary (final)" if final else "warnings summary"
-            self.write_sep("=", title, yellow=True, bold=False)
-            for message, warning_reports in reports_grouped_by_message.items():
-                has_any_location = False
+            def _collapsed_location_report(reports):
+                locations = []
                 for w in warning_reports:
                     location = w.get_location(self.config)
                     if location:
-                        self._tw.line(str(location))
-                        has_any_location = True
-                if has_any_location:
+                        locations.append(location)
+
+                if len(locations) < 10:
+                    return "\n".join(map(str, locations))
+
+                counts_by_filename = collections.Counter(
+                    str(loc).split("::", 1)[0] for loc in locations
+                )
+                return "\n".join(
+                    "{}: {} item{} warning with".format(k, v, "s" if v > 1 else "")
```

Comment:
At this point we will always have more than 1 item, so you don't need the logic to use proper plural here.

Reply:
It's required as there may be files with one item

Reply:
I found the "test-warns-collapse.py: 120 items warning with" text a bit weird to understand.

What about "test-warns-collapse.py (120 locations):" or so?